Lib/test/test_winconsoleio.py

```diff
@@ -92,9 +92,11 @@ def test_open_name(self):
         f.close()
         f.close()
 
-        f = open('C:/con', 'rb', buffering=0)
-        self.assertIsInstance(f, ConIO)
-        f.close()
+        # Windows 11 changed MS-DOS device name handling
+        if sys.getwindowsversion()[:3] < (10, 0, 22000):
+            f = open('C:/con', 'rb', buffering=0)
```

Comment:
What does `nt._getfullpathname('C:/con').lower()` return in Windows 11? The test should be skipped if it's not `r"\\.\con"`. Similarly, if `nt._getfullpathname(conout_path).lower()` isn't `r"\\.\conout$"`, assume that `conout_path` is a regular file.

Reply:
Both calls return a normalized pathname, not the device object name.

That said, as the tests in question are originally from you, shouldn't the tests then be written in terms of _getfullpathname() ?

Reply:
Meaning, instead of using open(), just check the results of _getfullpathname()

Reply:
It's checking `_PyIO_get_console_type()` in "Modules/_io/winconsoleio.c", which may need to be modified. Please also check `nt._getfullpathname()` and `nt._getfinalpathname()` (may fail) on the unqualified base names "con", "conin$", and "conout$". These names resolve to the console in previous Windows versions, so `_PyIO_get_console_type()` skips calling `GetFullPathNameW()`. This optimization may need to be removed if the API no longer special cases the base names.

Reply:
I verified the new behavior in Windows 11. If the path is qualified (i.e. contains slashes), only the "nul" device name appears to be reserved (e.g. "./nul" -> "\\\\.\\nul"). The other DOS device names aren't reserved in a qualified path (e.g. "./com1" is just a regular file in the current working directory). DOS device names are still reserved in an unqualified relative path, but extensions are no longer ignored (e.g. "con.txt" is just a regular file).

This means that `_PyIO_get_console_type()` doesn't need to be modified. It only special cases an exact case-insensitive match of "con", "conin$", and "conout$", which are still reserved paths.

Reply:
Sooooo..... does this mean that these changes are suitable?  Does it just need an OK from @zooba or @zware?  I'm only slightly impatient here as this is blocking the new buildbot from being actually useful.